platformio.ini

```diff
@@ -1358,13 +1358,29 @@ build_unflags       = ${common_stm32.build_unflags} -DUSBCON -DUSBD_USE_CDC -DUS
 extends              = lerdge_common
 board_build.firmware = Lerdge_X_firmware_force.bin
 
+#
+# Lerdge X with USB Flash Drive Support
+#
+[env:LERDGEX_usb_flash_drive]
+extends           = LERDGEX
+platform_packages = ${stm32_flash_drive.platform_packages}
+build_flags       = ${stm32_flash_drive.build_flags}
```

Comment:
This doesn't seem right to me. I know you're modeling it after the earlier board, but they don't seem right either.

If I understand how `extends` works properly, this will eliminate any board-specific defines from the `LERDGEX` environment. You would need to use something like this:

`build_flags       = ${env:LERDGEX.build_flags} ${stm32_flash_drive.build_flags}`

This has the downside that it might add unwanted flags, in which cause there might have to be a `stm32_flash_drive.build_unflags`.

@rhapsodyv can you confirm (or debunk) my thoughts?

Reply:
You can actually extend from multiple parents, but I don't know what will happen if they _both_ define `build_flags`.

It would be nice to be able to simply use something like this, but it would only work if their shared variables are merged, rather than one "winning".

`extends = common_stm32, stm32_flash_drive`

Reply:
I think I found the PIO code that implements extend. If I'm reading everything correctly it seems it will use the first value it encounters and ignores the rest:
https://github.com/platformio/platformio-core/blob/5a1b0e19b2b534d6faedb827bffc7ddb3f33479d/platformio/project/config.py#L210

Reply:
Correct. The `extends` keyword specifies the "base" for your current environment, and then the environment's settings replace the ones from the base. So, you can mix and match on each item, but not mix whole environments.